pkg/controller/vsphere/reconciler.go

```diff
@@ -333,6 +336,24 @@ func (r *Reconciler) reconcileNetwork(vm *virtualMachine) error {
 	return nil
 }
 
+func (r *Reconciler) reconcilePowerStateAnnontation(vm *virtualMachine) {
+	if vm == nil {
+		return
+	}
+
+	// This can return an error if machine is being deleted
+	powerState, err := vm.getPowerState()
+	if err != nil {
+		klog.V(3).Infof("%s: Failed to get power state: %v", r.machine.Name, err)
+		return
```

Comment:
why are we ignoring errors here?
we should return error, then is up to the consumer what to do with it, e.g log and ignore it if there's a reason to do so